File: scrapers/sources/aic.pl
{
	searchURL => 'http://www.artic.edu/aic/collections/search-artwork/results/keyword%3Awoodblock?page=%s',
	searchStart => 0,
	searchEnd => 225,
	searchIncrement => 1,

	urlExtractor => scraper {
		process 'span.italic a', 'urls[]' => '@href';
	},

	cleanURL => sub {
		my $url = shift;
		return $url =~ /(artwork\/\d+)/ ?
			"http://www.artic.edu/aic/collections/$1" :
			"";
	},

	extractPageID => qr/artwork\/(\d+)/,

	pageExtractor => scraper {
		process "id('tombstone')/p/a/child::text()", artist => 'TEXT';
		process "#tombstone .italic", title => 'TEXT';
		process "id('tombstone')/p/child::text()", date => 'TEXT';
		process "#tombstone p:nth-child(3)", description => 'TEXT';
		process "a.enlargement-link", source_image => '@href';
	},

	cleanPage => sub {
		my $item = shift;

		$item->{'source_url'} = "http://www.artic.edu/aic/collections/artwork/" . $item->{'source_id'};
		$item->{'date'} =~ s/^,\s*//;

		return $item;
	},

	extractImageFile => qr/([^\/]+.jpg)$/
}

File: scrapers/images/aic.js
exports.extract = {
	artist: "#tombstone/p/a/child::text()[position()=1]",
	title: "#tombstone//.italic",
	date: [ "#tombstone/p/child::text()", function( text ) {
		return text.replace( /^,\s*/, "" );
	} ],
	description: "#tombstone//p[position()=3]",
	source_image: "a.enlargement-link/@href || #artwork-image/img/@src"
};

exports.genURL = function( id ) {
	return "http://www.artic.edu/aic/collections/artwork/" + id;
};
